File: study-algorithm/src/nedaair/study/graph/dfs.py
__author__ = 'nedaair'

vList = [[2,3,4,6], [1,3], [1,2,5], [1,6], [3,6], [1,4,5]]

v = [1,2,3,4,5,6]

visited = []



def dfs() :

    for i in v :
        visited.append(False)

    for i in v :
        if(visited[i-1] == False) : aDfs(i)


def aDfs(x) :
    visited[x-1] = True

    for i in vList[x-1] :
        if(visited[i-1] == False) : aDfs(i)



dfs()

print visited

File: study-algorithm/src/xenomity/examination/graph/DFS.java
package xenomity.examination.graph;

public class DFS implements Sortable {

	@Override
	public void sort(int[][] matrixData, int beginVertexIndex) {
		boolean[] visited = new boolean[matrixData.length];
		
		aDfs(matrixData, beginVertexIndex, visited);
	}
	
	private void aDfs(int[][] matrixData, int vertexIndex, boolean[] visited) {
		visited[vertexIndex] = true;
		
		System.out.printf("%d ", vertexIndex);
		
		for (int i = 0; i < matrixData[vertexIndex].length; i++) {
			int linkedVertex = matrixData[vertexIndex][i];
			
			if (!visited[linkedVertex]) {
				aDfs(matrixData, linkedVertex, visited);
			}
		}
	}

}
